pkg/process/v6/transformers/osv/transform.go

```diff
@@ -0,0 +1,299 @@
+				Assigners:   nil,
+				Description: vulnerability.Details,
+				References:  getReferences(vulnerability),
+				Aliases:     vulnerability.Aliases,
+				Severities:  severities,
+			},
+		},
+	}
+
+	for _, a := range getAffectedPackages(vulnerability) {
+		in = append(in, a)
+	}
+
+	return transformers.NewEntries(in...), nil
+}
+
+func getAffectedPackages(vuln unmarshal.OSVVulnerability) []grypeDB.AffectedPackageHandle {
+	if len(vuln.Affected) == 0 {
+		return nil
+	}
+
+	// CPES might be in the database_specific information
+	cpes, withCPE := vuln.DatabaseSpecific["cpes"]
+	if withCPE {
+		if _, ok := cpes.([]string); !ok {
+			withCPE = false
+		}
+	}
+
+	var aphs []grypeDB.AffectedPackageHandle
+	for _, affected := range vuln.Affected {
+		aph := grypeDB.AffectedPackageHandle{
+			Package:   getPackage(affected.Package),
+			BlobValue: &grypeDB.AffectedPackageBlob{CVEs: vuln.Aliases},
+		}
+
+		if withCPE {
+			aph.BlobValue.Qualifiers = &grypeDB.AffectedPackageQualifiers{
+				PlatformCPEs: cpes.([]string),
+			}
+		}
+
+		var ranges []grypeDB.AffectedRange
+		for _, r := range affected.Ranges {
+			ranges = append(ranges, getGrypeRangesFromRange(r)...)
+		}
+		aph.BlobValue.Ranges = ranges
+		aphs = append(aphs, aph)
+	}
+
+	// stable ordering
+	sort.Sort(internal.ByAffectedPackage(aphs))
+
+	return aphs
+}
+
+// OSV supports flattered ranges, so both formats below are valid:
+// "ranges": [
+//
+//	{
+//	  "type": "SEMVER",
+//	  "events": [
+//	    {
+//	      "introduced": "12.0.0"
+//	    },
+//	    {
+//	      "fixed": "12.18.4"
+//	    }
+//	  ]
+//	},
+//	{
+//	  "type": "SEMVER",
+//	  "events": [
+//	    {
+//	      "introduced": "14.0.0"
+//	    },
+//	    {
+//	      "fixed": "14.11.0"
+//	    }
+//	  ]
+//	}
+//
+// ]
+// "ranges": [
+//
+//	{
+//	  "type": "SEMVER",
+//	  "events": [
+//		{
+//		  "introduced": "12.0.0"
+//		},
+//		{
+//		  "fixed": "12.18.4"
+//		},
+//		{
+//		  "introduced": "14.0.0"
+//		},
+//		{
+//		  "fixed": "14.11.0"
+//		}
+//	  ]
+//	}
+//
+// ]
+func getGrypeRangesFromRange(r models.Range) []grypeDB.AffectedRange {
+	var ranges []grypeDB.AffectedRange
+	if len(r.Events) == 0 {
+		return nil
+	}
+
+	rangeType := normalizeRangeType(r.Type)
+	var constraint string
+	for _, e := range r.Events {
+		if e.Introduced != "" {
+			constraint = fmt.Sprintf("%s >=", e.Introduced)
+		} else if e.LastAffected != "" {
+			eventConstraint := fmt.Sprintf("<= %s", e.LastAffected)
+			if constraint == "" {
+				constraint = eventConstraint
+			} else {
+				constraint = common.AndConstraints(constraint, eventConstraint)
+			}
+			// We don't know the fix if last affected is set
+			ranges = append(ranges, grypeDB.AffectedRange{
+				Version: grypeDB.AffectedVersion{
+					Type:       rangeType,
+					Constraint: normalizeConstraint(constraint, rangeType),
+				},
+			})
+			// Reset the constraint
+			constraint = ""
+		} else if e.Fixed != "" {
+			eventConstraint := fmt.Sprintf("<= %s", e.Fixed)
```

Comment:
This should be `<`, not `<=`.

Reply:
done at https://github.com/anchore/grype-db/pull/217/commits/d8b70c7911c2dc858c1f42b59f285af49e5ac3ca